### PR TITLE
Add attached databases#7208

### LINKED ISSUE #7208: Support for attached databases (SQLite)
I can attach a sqlite db from the sql editor using ATTACH DATABASE org AS someName.

But in the connections pane the tables in the attached DB are not shown in the current connection.

Perhaps would be of general interest that these tables were shown and you have the time to add this new feature.

Thanks in advance

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes #7208

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java [hunk 1]
```diff
@@ -171,15 +182,22 @@ public boolean supportsSequences(@NotNull GenericDataSource dataSource) {
...
+    ) throws SQLException {
+        return super.prepareForeignKeysLoadStatement(session, owner, forParent);
+    }
+
     @Override
     public JDBCStatement prepareSequencesLoadStatement(@NotNull JDBCSession session, @NotNull GenericStructContainer container) throws SQLException {
         // The sqlite_sequence table is created and initialized automatically whenever a normal table that contains an AUTOINCREMENT column is created. Not earlier
         try {
-            JDBCUtils.queryString(session, "SELECT 1 FROM sqlite_sequence");
+            JDBCUtils.queryString(session, "SELECT 1 FROM" + getFullyQualifiedName(container, "sqlite_sequence"));
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]:
```suggestion
            JDBCUtils.queryString(session, "SELECT 1 FROM " + getFullyQualifiedName(container, "sqlite_sequence"));
```

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java [hunk 2]
```diff
@@ -171,15 +182,22 @@ public boolean supportsSequences(@NotNull GenericDataSource dataSource) {
...
         } catch (SQLException e) {
             throw new SQLException("Error loading SQLite sequences. Probably sqlite_sequence info table doesn't exist yet. Please create table with AUTOINCREMENT column first.", e);
         }
-        return session.prepareStatement("SELECT * FROM sqlite_sequence");
+        return session.prepareStatement("SELECT * FROM"  + getFullyQualifiedName(container, "sqlite_sequence"));
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]:
```suggestion
        return session.prepareStatement("SELECT * FROM "  + getFullyQualifiedName(container, "sqlite_sequence"));
```

**Review by E1izabeth — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java [hunk 3]
```diff
@@ -171,15 +182,22 @@ public boolean supportsSequences(@NotNull GenericDataSource dataSource) {
         return true;
     }
 
+    @Override
+    public JDBCStatement prepareForeignKeysLoadStatement(
+        @NotNull JDBCSession session, @NotNull GenericStructContainer owner, @Nullable GenericTableBase forParent
+    ) throws SQLException {
+        return super.prepareForeignKeysLoadStatement(session, owner, forParent);
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: Any specific reason for this code here?

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java [hunk 4]
```diff
@@ -74,6 +74,11 @@ public String getTableDDL(DBRProgressMonitor monitor, GenericTableBase sourceObj
         return tableDDL + "\n" + indexesDDL;
     }
 
+    @Override
+    public boolean useCatalogInObjectNames() {
+        return super.useCatalogInObjectNames();
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: Any specific reason for this code here?

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: migration artifact, will be removed

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java (hunk 3 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: migration artifact, will be removed

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java [hunk 5]
```diff
@@ -40,12 +41,24 @@
...
     }
 
+    public SQLiteDataSource(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull DBPDataSourceContainer container,
+        @NotNull GenericMetaModel metaModel,
+        @NotNull SQLDialect dialect
+    ) throws DBException {
+        super(monitor, container, metaModel, dialect);
+    }
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java:59]: Do you need it?

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java (hunk 3 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteMetaModel.java]: it's still here

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java (hunk 5 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java:59]: This one is needed. See it's extension.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java:59]: Which extension are you talking about? Please, point me out.

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java:59]: EE part

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Pull request merged**